src/lib/use-color-theme.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react';
+import { classes } from '../components/app.st.css'
+
+export function useColorTheme() {
+  const [theme, setTheme] = useState<'dark' | 'light'>('dark')
```

Comment:
the initial value here should be a condition on `window.matchMedia('(prefers-color-scheme: dark)').matches`
this will save the initial re-render when this hook is applied and also prevent color jump.